File: problema02/sol_python/problema02.py
cantidad = int(input("Ingrese la cantidad de trajes: "))
precio = float(input("Ingrese el precio unitario de los trajes: $"))
subtotal = cantidad * precio
descuento = 0
total = subtotal
if cantidad == 1:
    descuento = (subtotal * 20) / 100
    total = subtotal - descuento
elif cantidad == 2:
    descuento = (subtotal * 25) / 100
    total = subtotal - descuento
elif cantidad == 3:
    descuento = (subtotal * 40) / 100
    total = subtotal - descuento
elif cantidad > 3:
    descuento = (subtotal * 50) / 100
    total = subtotal - descuento
print(f"El subtotal por pagar es: ${subtotal}\nEl descuento es: ${descuento}\nEl total por pagar es: "
      f"${total}")

File: problema02/sol_java/problema02/src/problema02/Problema02.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Main.java to edit this template
 */
package problema02;

import java.util.Locale;
import java.util.Scanner;

/**
 *
 * @author renat
 */
public class Problema02 {

    /**
     * @param args the command line arguments
     */
    public static void main(String[] args) {
        // TODO code application logic here
        Scanner entrada = new Scanner(System.in);
        entrada.useLocale(Locale.US);
        
        int cantidad;
        double precio;
        double subtotal;
        double descuento;
        double total;
        
        System.out.println("Ingrese la cantidad de trajes:");
        cantidad = entrada.nextInt();
        
        System.out.println("Ingrese el precio unitario de los trajes:");
        precio = entrada.nextDouble();
        
        subtotal = cantidad * precio;
        total = subtotal;
        descuento = 0;
        
        if (cantidad == 1) {
            descuento = (subtotal * 20) / 100;
            total = subtotal - descuento;
        } else {
            if (cantidad == 2) {
                descuento = (subtotal *25) / 100;
                total = subtotal - descuento;
            } else {
                if (cantidad == 3) {
                    descuento = (subtotal * 40) / 100;
                    total = subtotal - descuento;
                } else {
                    if (cantidad > 3) {
                        descuento = (subtotal * 50) / 100;
                        total = subtotal - descuento;
                    }
                }
            }
        }
        System.out.printf ("El subtotal por pagar es: $%.2f\nEl descuento "
                + "es: $%.2f\nEl total por pagar es: $%.2f\n", subtotal, 
                descuento, total);
    }
    
}
